src/compiler/checker.ts

```diff
@@ -4151,13 +4151,13 @@ namespace ts {
             if (parentType === unknownType) {
                 return unknownType;
             }
-            // If no type was specified or inferred for parent, or if the specified or inferred type is any,
-            // infer from the initializer of the binding element if one is present. Otherwise, go with the
-            // undefined or any type of the parent.
-            if (!parentType || isTypeAny(parentType)) {
-                if (declaration.initializer) {
-                    return checkDeclarationInitializer(declaration);
-                }
+            // If no type was specified or inferred for parent,
+            // infer from the initializer of the binding element if one is present.
+            // Otherwise, go with the undefined type of the parent.

```

Comment:
Doesn't it make sense to say "Otherwise, go with the any type of the parent"? We don't seem to do any checking for the `undefined` type here?

Reply:
This applies to the block immediately below the comment, which checks `if (!parentType)`. "The falsy type of the parent" would be more accurate, but isn't the intent.

Any is different in that (1) you can specify any and (2) it will override the initialiser. I decided not to comment this as it seemed pretty obvious.